### ISSUE TITLE
Function type conversion mistake?

### ISSUE THREAD

**Issue by emamoah** [ethereum/solidity#13942]
## Page

https://docs.soliditylang.org/en/v0.8.18/types.html#function-types

## Abstract

Under the sub-section "**Conversions**", it's mentioned that a function type of **A** is convertible to **B**, if **A** is _more restrictive_ than **B**.
In the examples, it says `payable` is convertible to `non-payable`. So taking `A = payable`, and `B = non-payable`, it looks contradictory in the next paragraph, which says that `non-payable` functions (B) are more restrictive than `payable` ones (A). Is this a mistake, or I'm just not getting it right?

## Pull request

No pull request made, since I'm not sure if my argument is right.

**Comment by ekpyron** [comment]
Does the paragraph specifically about this potentially being confusing help :-)?

**Comment by cameel** [comment]
Related: #13114/#13613.

**Comment by emamoah** [comment]
@ekpyron Not exactly. That part talks about the mechanism of `payable` and `non-payable`.
My question isn't about the convertibility of `payable` to `non-payable`. That definitely makes sense. I understand it.
My problem however is in this paragraph:

> A function type `A` is implicitly convertible to a function type `B` if and only if [...] the state mutability of `A` is _more restrictive_ than the state mutability of `B`. In particular: [...] `payable` functions can be converted to `non-payable` functions.

This is completely contradictory from my point of view. That's why I did the comparison of `A = payable` and `B = non-payable`.
The first part says `A` can be converted to `B` if `A` is _more restrictive_. But `A`, which is `payable` is supposed to be _less restrictive_ than `non-payable`.

Am I getting it wrong?

**Comment by daredevil3435** [comment]
@emamoah you're right! non-payable is more restrictive than payable.I'm interested in contributing this issue.
